hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountBalanceHandler.java

```diff
@@ -87,7 +87,9 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         } else if (op.hasContractID()) {
             final var contract = accountStore.getContractById(requireNonNull(op.contractID()));
             validateFalsePreCheck(contract == null || !contract.smartContract(), INVALID_CONTRACT_ID);
-            validateFalsePreCheck(contract.deleted(), CONTRACT_DELETED);
+            if (contract != null) {
+                validateFalsePreCheck(contract.deleted(), CONTRACT_DELETED);
```

Comment:
Doesn't hurt, but if contract is null, we will already throw exception in line 89 right ? Probably Sonar still shows this as error ? I don't think this needs to be fixed

Reply:
Yes, line 89 will throw an exception anyway. I was debating between fixing it or not. I will revert the change 👍